src/Sylius/Bundle/WebBundle/Controller/Frontend/Review/ProductReviewController.php

```diff
@@ -0,0 +1,163 @@
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * ProductReview controller.
+ *
+ * @author Justin Hilles <justin@1011i.com>
+ * @author Daniel Richter <nexyz9@gmail.com>
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class ProductReviewController extends ResourceController
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function indexAction(Request $request)
+    {
+        $product = $this->findProductOr404();
+
+        $criteria = $request->query->get('criteria', array());
+        $criteria['product'] = $product;
+        $request->query->set('criteria', $criteria);
+
+        return parent::indexAction($request);
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function createAction(Request $request)
+    {
+        $this->isGrantedOr403('create');
+
+        $resource = $this->createNew();
+        $form = $this->getForm($resource);
+
+        $view = $this
+            ->view()
+            ->setTemplate($this->config->getTemplate('create.html'))
+            ->setData(array(
+                $this->config->getResourceName() => $resource,
+                'form'                           => $form->createView(),
+            ))
+        ;
+
+        return $this->handleView($view);
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function createWithAjaxAction(Request $request)
+    {
+        $resource = $this->createNew();
+        $form = $this->getForm($resource);
+
+        if (!$this->createResourceIfValid($request, $form)) {
+            return new JsonResponse($this->getDeepErrors($form));
+        }
+
+        return new JsonResponse('success');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createNew()
+    {
+        $review = parent::createNew();
+        $review->setProduct($this->findProductOr404());
+
+        return $review;
+    }
+
+    /**
+     * @return ProductInterface
+     */
+    private function findProductOr404()
+    {
+        $slug = $this->getRequest()->get('slug');
+
+        if (!$product = $this->get('sylius.repository.product')->findOneBy(array('slug' => $slug))) {
+            throw new NotFoundHttpException(sprintf('Product with slug "%s" not found', $slug));
+        }
+
+        return $product;
+    }
+
+    /**
+     * @param Request $request
+     * @param FormInterface $form
+     *
+     * @return bool
+     */
+    private function createResourceIfValid(Request $request, FormInterface $form)
+    {
+        if ($form->submit($request)->isValid()) {
+            $this->domainManager->create($form->getData());
+
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param FormInterface $form
+     *
+     * @return array
+     */
+    private function getDeepErrors(FormInterface $form)
```

Comment:
If it's scheduled for `0.16` Sylius version, you can use `$form->getErrors(true)` instaed of this method.